lib/rack/lint.rb

```diff
@@ -758,21 +758,53 @@ def verify_content_length(size)
       ## that yields +String+ instances, a +Proc+ instance, or a File-like
       ## object.
       ##
-      ## The Body must respond to +each+ or +call+. It may optionally respond to
-      ## +to_path+.
+      ## The Body must respond to +each+ or +call+. It may optionally respond
+      ## to +to_path+ or +to_ary+. A Body that responds to +each+ is considered
+      ## to be an Enumerable Body. A Body that responds to +call+ is considered
+      ## to be a Streaming Body.
       ##
-      ## A Body that responds to +each+ is considered to be an Enumerable Body.
-      ##
-      ## A Body that responds to +call+ is considered to be a Streaming Body.
-      ##
-      ## A Body that responds to +to_path+ is expected to generate the same
-      ## content as would be produced by reading a local file opened with the
-      ## path returned by calling +to_path+.
-      ##
-      ## A body that responds to both +each+ and +call+ must be treated as an
+      ## A Body that responds to both +each+ and +call+ must be treated as an
       ## Enumerable Body, not a Streaming Body. If it responds to +each+, you
-      ## must call +each+ and not +call+. If the body doesn't respond to
+      ## must call +each+ and not +call+. If the Body doesn't respond to
       ## +each+, then you can assume it responds to +call+.
+      ##
+      ## The Body must either be consumed or returned. The Body is consumed by
+      ## optionally calling one of +each+, +call+, +to_path+, or +to_ary+.
+      ## Then, if the Body responds to +close+, it must be called to release
+      ## any resources associated with the generation of the body.
+      def close
+        ##
+        ## After calling +close+, the Body is considered closed and should not
+        ## be consumed again.
+        return if @closed
```

Comment:
You are right, but are we comfortable executing all the validation logic multiple times?

I don't have a strong opinion about this but I think it's easier if we allow it to be called multiple times. But it also feels a bit wrong to me to do that.

Reply:
I think we should at least call `@body.close` every time `close` is called.  I'm fine if you don't want to run the `rack.lint` check multiple times.

Reply:
I’m just kind of worried the logic of the check might break if it’s called at the wrong times. If it’s called too early it might not work because the stack of all lint instances I assume can grow… I will check it in more detail.